File: 3de year project/Woodrock-Animal-Rescue-master/BLL/Medication.cs
using System;
using System.Collections.Generic;
using System.Text;
using Woodrock_Animal_Rescue.DAL;


namespace Woodrock_Animal_Rescue.BLL
{
    public class Medication
{

        //Will add comments latyer - Hanru Bosch aka da beast 

        #region Properties

        private int medicationId;
        private string medicationName;
        private float stock;
        private float cost;
        private string description;
        private Array diagnosises;

        public int MedicationId { get => medicationId; set => medicationId = value; }
        public string MedicationName { get => medicationName; set => medicationName = value; }
        public float Stock { get => stock; set => stock = value; }
        public float Cost { get => cost; set => cost = value; }
        public string Description { get => description; set => description = value; }
        public Array Diagnosises { get => diagnosises; set => diagnosises = value; }



        #endregion


        #region Constructor

        public Medication(int medicationId, string medicationName, float stock, float cost, string description, Array diagnosises)
        {
            MedicationId = medicationId;
            MedicationName = medicationName;
            Stock = stock;
            Cost = cost;
            Description = description;
            Diagnosises = diagnosises;
        }

        public Medication()
        {

        }

        #endregion


        #region Methods & Functions

        public override string ToString()
        {
            return string.Format("Medication Name: {0}, Stock: {1}, Cost: R{2}, Description: {3}, Diagnosises: {4}", medicationName, stock, cost, description, diagnosises);
        }

        public void AddMedication(int medicationId, string medicationName, float stock, float cost, string description, Array diagnoses)
        {
            string qry = string.Format("INSERT INTO Animal_Medication (MedicationID,MedicationName,Stock,Cost,Description,Diagnosises)  VALUES ('{0}','{1}','{2}','{3}','{4}','{5}') ", medicationId, medicationName, stock, cost, description, diagnoses);
            DataHandler dh = new DataHandler();
            dh.Insert(qry);
        }

        public void IncreaseStock(int amount, int medicationID)
        {
            string qry = string.Format("UPDATE Animal_Medication SET Stock = '{0}' WHERE MedicationID = {1}", amount, medicationID);
            DataHandler dh = new DataHandler();
            dh.Update(qry);
        }

        public void DecreaseStock(int amount, int medicationID)
        {
            string qry = string.Format("UPDATE Animal_Medication SET Stock = '{0}' WHERE MedicationID = {1}", amount, medicationID);
            DataHandler dh = new DataHandler();
            dh.Update(qry);
        }

        public void EditDetails(float cost, string description, int medicationID)
        {
            string qry = string.Format("UPDATE Animal_Medication SET Cost = '{0}', Description = '{1}' WHERE MedicationID = {2}", cost, description, medicationID);
            DataHandler dh = new DataHandler();
            dh.Update(qry);
        }

        public void AddDiagnosis(int medicationID, Array diagnosis)
        {
            string qry = string.Format("UPDATE Animal_Medication SET Diagnosis = '{0}' WHERE MedicationID = {1}", diagnosis, medicationID);
            DataHandler dh = new DataHandler();
            dh.Update(qry);
        }

        public void DeleteMedication(int medicationID)
        {
            string qry = string.Format("DELETE FROM Animal_Medication WHERE MedicationID = {0}", medicationID);
            DataHandler dh = new DataHandler();
            dh.Delete(qry);
        }

        #endregion



    }
}

File: 3de year project/Woodrock-Animal-Rescue-master/BLL/Medication.php
<?php
$servername = "localhost";
$dbname = "WoodRock";

$conn = new mysqli($servername, $username, $password, $dbname);
if ($conn->connect_error) 
{
    die("Connection failed: " . $conn->connect_error);
}

class Medication
{
    public $medicationId;
    public $medicationName;
    public $stock;
    public $cost;
    public $description;
    public $diagnosises;

    function _constructor($medicationId, $medicationName, $stock, $cost, $description, $diagnosises)
    {
        $this ->medicationId = $medicationId;
        $this ->medicationName = $medicationName;
        $this ->stock = $stock;
        $this ->cost = $cost;
        $this ->description = $description;
        $this ->diagnosises = $diagnosises;
    }

    function _AddMedication($medicationId, $medicationName, $stock, $cost, $description, $diagnosises)
    {
        $sql = ("INSERT INTO Animal_Medication (MedicationID,MedicationName,Stock,Cost,Description,Diagnosises)  VALUES ('{0}','{1}','{2}','{3}','{4}','{5}') ", $medicationId, $medicationName, $stock, $cost, $description, $diagnosises);

        if ($conn->query($sql) === TRUE) {
            echo "New record created successfully";
        } else {
            echo "Error: " . $sql . "<br>" . $conn->error;
        }
    }

    function _IncreaseStock($amount, $medicationID)
    {
        $sql = ("UPDATE Animal_Medication SET Stock = '{0}' WHERE MedicationID = '{1}'", $amount, $medicationID);

        if ($conn->query($sql) === TRUE) {
            echo "Record updated successfully";
        } else {
            echo "Error updating record: " . $conn->error;
        }        
    }

    function _DecreaseStock($amount, $medicationID)
    {
        $sql = ("UPDATE Animal_Medication SET Stock = '{0}' WHERE MedicationID = '{1}'", $amount, $medicationID);

        if ($conn->query($sql) === TRUE) {
            echo "Record updated successfully";
        } else {
            echo "Error updating record: " . $conn->error;
        }    
    }

    function _EditDetails($cost, $description, $medicationID)
    {
        $sql = ("UPDATE Animal_Medication SET Cost = '{0}', Description = '{1}' WHERE MedicationID = '{2}'", cost, description, medicationID);
        if ($conn->query($sql) === TRUE) {
            echo "Record updated successfully";
        } else {
            echo "Error updating record: " . $conn->error;
        }   
    }

    function _AddDiagnosis($diagnosis, $medicationID)
    {
        $sql = ("UPDATE Animal_Medication SET Diagnosis = '{0}' WHERE MedicationID = '{1}'", $diagnosis, $medicationID);
        if ($conn->query($sql) === TRUE) {
            echo "Record updated successfully";
        } else {
            echo "Error updating record: " . $conn->error;
        }  
    }

    function _DeleteMedication($medicationID)
    {
        $sql = ("DELETE FROM Animal_Medication WHERE MedicationID = '{0}'", $medicationID)

        if (mysqli_query($conn, $sql)) {
            echo "Record deleted successfully";
        } else {
            echo "Error deleting record: " . mysqli_error($conn);
        }
    }
}
